x/data/iri.go

```diff
@@ -0,0 +1,103 @@
+package data
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/btcsuite/btcutil/base58"
+)
+
+// AccAddressToDID converts an account address to a DID using a chain-specific method prefix,
+// which should generally be equivalent to the chain's bech32 account prefix.
+func AccAddressToDID(address sdk.AccAddress, methodPrefix string) string {
+	hash := base58.CheckEncode(address, didVersion0)
+	return fmt.Sprintf("did:%s:%s", methodPrefix, hash)
+}
+
+func (ch ContentHash) ToIRI() (string, error) {
+	switch hash := ch.Sum.(type) {
+	case *ContentHash_Raw_:
+		return hash.Raw.ToIRI()
+	case *ContentHash_Graph_:
+		return hash.Graph.ToIRI()
+	default:
+		return "", fmt.Errorf("invalid %T type %T", ch, hash)
+	}
+}
+
+const (
+	iriVersion0 byte = 0
+
+	IriPrefixRaw   byte = 0
+	IriPrefixGraph byte = 1
+
+	didVersion0 byte = 0
+)
+
+func (chr ContentHash_Raw) ToIRI() (string, error) {
+	err := chr.Validate()
+	if err != nil {
+		return "", err
+	}
+
+	bz := make([]byte, len(chr.Hash)+2)
+	bz[0] = IriPrefixRaw
+	bz[1] = byte(chr.DigestAlgorithm)
+	copy(bz[2:], chr.Hash)
+	hashStr := base58.CheckEncode(bz, iriVersion0)
+
+	ext, err := chr.MediaType.ToExtension()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("regen:%s.%s", hashStr, ext), nil
```

Comment:
well in this case I think we actually want the full format string because it gets compiled to a constant whereas doing something like `RegenConstant + "%s.%s"` leads to extra memory at runtime. does that make sense? or were you talking about something different?